src/shared/database/index.ts

```diff
@@ -482,6 +460,46 @@ class Pindexer {
       .where('transaction_id', '=', Buffer.from(hexToUint8Array(txHash)))
       .executeTakeFirst();
   }
+
+  async queryLeaderboard(limit: number, interval: string, baseHex?: string, quoteHex?: string) {
+    const positionExecutions = this.db
+      .selectFrom('dex_ex_position_executions')
+      .select(exp => [
+        'position_id',
+        'context_asset_start',
+        'context_asset_end',
+        sql<number>`sum(${exp.ref('delta_1')} + ${exp.ref('lambda_1')})`.as('volume1'),
+        sql<number>`sum(${exp.ref('delta_2')} + ${exp.ref('lambda_2')})`.as('volume2'),
+        sql<number>`sum(${exp.ref('fee_1')})`.as('fees1'),
+        sql<number>`sum(${exp.ref('fee_2')})`.as('fees2'),
+        sql<number>`CAST(count(*) AS INTEGER)`.as('executionCount'),
+      ])
+      .$if(baseHex !== undefined, qb =>
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- baseHex is defined
+        qb.where('context_asset_start', '=', Buffer.from(hexToUint8Array(baseHex!))),
+      )
+      .$if(quoteHex !== undefined, qb =>
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- quoteHex is defined
+        qb.where('context_asset_end', '=', Buffer.from(hexToUint8Array(quoteHex!))),
```

Comment:
thought/suggestion: the `quoteHex` and `baseHex` arguments are defined as optional, hence the fair eslint warning. Can these arguments be made required or maybe checked for undefined?